sdk/java/src/main/java/com/chain/http/Client.java

```diff
@@ -379,19 +379,24 @@ public void setProxy(Proxy proxy) {
   }
 
   private OkHttpClient buildHttpClient(Builder builder) {
-    OkHttpClient httpClient = new OkHttpClient();
+    OkHttpClient httpClient = builder.httpClient.clone();
+    httpClient.setFollowRedirects(false);
 
     if (builder.sslSocketFactory != null) {
       httpClient.setSslSocketFactory(builder.sslSocketFactory);
     }
-
-    httpClient.setFollowRedirects(false);
-    httpClient.setReadTimeout(builder.readTimeout, builder.readTimeoutUnit);
-    httpClient.setWriteTimeout(builder.writeTimeout, builder.writeTimeoutUnit);
-    httpClient.setConnectTimeout(builder.connectTimeout, builder.connectTimeoutUnit);
-
-    httpClient.setConnectionPool(builder.pool);
-
+    if (builder.readTimeoutUnit != null) {
```

Comment:
Discussed out of band: `Client#buildHttpClient` sets a couple of parameters on the httpClient that require an accumulation of dependent properties, i.e. we have to set a logger and log level before we can add a request interceptor. In the future, we will need to set trusted ca’s and client certs before we can build the SslSocketFactory. In lieu of adding explicit create calls for Builder fields, we will keep this as is and update the name of `Builder#httpClient` to `Builder#baseHttpClient` so it's clear that the field is a base that can be set and not the actual object the builder setters are operating upon.